src/ripple/app/hook/Enum.h

```diff
@@ -0,0 +1,323 @@
+#include <set>
+#include <string>
+#ifndef HOOKENUM_INCLUDED
+#define HOOKENUM_INCLUDED 1
+namespace ripple
+{
+    enum HookSetOperation : int8_t
+    {
+        hsoINVALID  = -1,
```

Comment:
I spent some time experimenting with replacing it with an `enum class` and `std::optional<HookSetOperation>` (with the std::nullopt replacing INVALID). It's a fairly large code change for relatively little pay-off, and until c++20 is enabled (e.g. allowing use of `using enum`) you need to fully specify each enumeral e.g. `HookSetOperation::Install` which is harder to read than `hsoINSTALL`. Additionally an extra branch is then needed above each switch statement to test if the optional came back populated rather than just a combined default/INVALID label at the bottom and inside the switch. Maybe this still has merit. Let's circle back when C++20 is enabled. For now leaving it as is.